From edfd2c0ba42a35e223b3bd35983edf9427c11fa8 Mon Sep 17 00:00:00 2001
From: ChonDoit <thphantomblog@gmail.com>
Date: Sat, 6 Aug 2022 15:20:41 +0000
Subject: [PATCH 12/16] Add offline Charge Service

Properly Thanks to Victor Bo/Ponces for base patches.

Additions
| Use default charger service instead gsicharger
| Capabilities SYS_BOOT to be able to directly boot system
| Seclabel before run service
| chown-chmod for Smart Charging
---
 rootdir/init.rc | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index ab45bb032..a26e6679e 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -11,6 +11,17 @@ import /vendor/etc/init/hw/init.${ro.hardware}.rc
 import /system/etc/init/hw/init.usb.configfs.rc
 import /system/etc/init/hw/init.${ro.zygote}.rc
 
+service charger /bin/charger
+    class charger
+    user system
+    group system shell graphics input wakelock
+    capabilities SYS_BOOT
+    seclabel u:r:charger:s0
+
+on boot
+    chown system system /sys/class/power_supply/batery/input_suspend
+    chmod 0777 /sys/class/power_supply/batery/input_suspend
+
 # Cgroups are mounted right before early-init using list from /etc/cgroups.json
 on early-init
     # Disable sysrq from keyboard
@@ -1187,6 +1198,10 @@ on nonencrypted
 on property:sys.init_log_level=*
     loglevel ${sys.init_log_level}
 
+on charger && property:ro.hardware=mt*
+    write /sys/class/leds/lcd-backlight/trigger "backlight"
+    write /sys/class/android_usb/android0/enable 1
+
 on charger
     class_start charger
 
-- 
2.34.1

